rules/generate_rules.py

```diff
@@ -13,12 +13,23 @@
 
 FUZZY_RULES = yaml.safe_load(rules_src.read_text())["fuzzyRules"]
 
+for rule in FUZZY_RULES:
+    if "name" not in rule:
+        print("Fuzzy rule is missing a name, exiting")
+        sys.exit()
```

Comment:
For the sake of it, you should exit with `1` or even more explicitly raise `SystemExit("FZ rule missing")`

Reply:
Agreed